src/streams/SubscriptionsScreen.js

```diff
@@ -60,7 +60,8 @@ export default function SubscriptionsScreen(props: Props): Node {
       .sort((a, b) => caseInsensitiveCompareFunc(a.name, b.name));
     return [
       { key: 'Pinned', data: sortedSubscriptions.filter(x => x.pin_to_top) },
-      { key: 'Unpinned', data: sortedSubscriptions.filter(x => !x.pin_to_top) },
+      { key: 'Unpinned', data: sortedSubscriptions.filter(x => !x.pin_to_top && x.in_home_view) },
+      { key: 'Muted', data: sortedSubscriptions.filter(x => !x.pin_to_top && !x.in_home_view) },
```

Comment:
Let's avoid going through the list of subscriptions three times here. We can do it with just one pass, by using a plain old loop instead of `.filter`.

That will also let us deduplicate the conditionals on `.pin_to_top` and `.in_home_view`.

With the old code it's a bit of a closer call between those advantages of a loop, and the declarative neatness of this `.filter` version. But when there's three filters instead of two, and the conditions get further duplicated between them like this, it's time to make it a loop.

Let's have one commit that's NFC (https://github.com/zulip/zulip-mobile/blob/main/docs/glossary.md#nfc ) and just changes the existing code to use a loop and make a single pass. Then a subsequent commit can make the substantive change here.